doc/man3/EVP_PKEY_set1_RSA.pod

```diff
@@ -62,13 +62,16 @@ see L<openssl_user_macros(7)>:
 EVP_PKEY_get_base_id() returns the type of I<pkey>. For example
 an RSA key will return B<EVP_PKEY_RSA>.
 
-EVP_PKEY_get_id() returns the actual OID associated with I<pkey>.
-Historically keys using the same algorithm could use different OIDs.
-For example an RSA key could use the OIDs corresponding to
+EVP_PKEY_get_id() returns the actual NID associated with I<pkey>
+if I<pkey> is not implemented in an external L<provider(7)>.
```

Comment:
Is that sufficient? What if it's implemented in an engine? What if implemented (only) in the FIPS provider?

Reply:
> `if I<pkey> type is a type (also) implemented in the default provider.` (or something like that)

That's actually incorrect.  If an algorithm was *only* implemented in the default provider, `EVP_PKEY_get_id()` would return `EVP_PKEY_KEYMGMT`.  We return other NIDs *because those algorithms are also implemented in form of `EVP_PKEY_ASN1_METHOD` (i.e. legacy) structures, in `libcrypto` itself*

Reply:
Yes, the problem is that might be even more confusing. I do not really have a good proposal for better wording but the current as is in this PR also isn't correct. `EVP_PKEY_get_id() returns the actual NID associated with I<pkey> if the type of the key was known to older versions of the OpenSSL library.` ? But that seems convoluted.

Reply:
What about this:

    EVP_PKEY_get_id() returns the actual NID associated with I<pkey> if I<pkey> is implemented outside of a L<provider(7)>. 

and vice versa below

    EVP_PKEY_get_id() returns EVP_PKEY_KEYMGMT if I<pkey> is only implemented within a L<provider(7)>. 

? --> Implemented in https://github.com/openssl/openssl/pull/20501/commits/a5ca039d0248101ad930096df9e6a69486a671fb